### PR TITLE
add WinForms sample

### PR DESCRIPTION
**Pull request by anurse**
Associated with #2008 

Adds a Windows Forms Client. I don't believe this will fail to build on macOS/Linux, but if it does the CI will tell us and we can adjust. The client doesn't work right now because of #2008 but it builds

image

### CODE REVIEW

File: samples/WindowsFormsSample/ChatForm.cs [hunk 1]
```diff
@@ -0,0 +1,132 @@
+using System;
```
Comment (JamesNK) [inline samples/WindowsFormsSample/ChatForm.cs]: Copyright header

File: samples/WindowsFormsSample/Program.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+using System;
```
Comment (JamesNK) [inline samples/WindowsFormsSample/Program.cs]: Copyright header.

**Comment by davidfowl** [comment]
When travis works I'll buy it

**Comment by anurse** [comment]
FYI this won't be merged until the branching is finished, and it'll be going into dev.

**Comment by anurse** [comment]
```
Could not run the "GenerateResource" task because MSBuild could not create or connect to a task host with runtime "CLR4" and architecture "x86".
```

☹️ 

I'll look at disabling this on non-Windows.

**Comment by anurse** [comment]
:up: :date: nits picked, and I think I found a sneaky way to make the build no-op on non-Windows.

**Comment by BrennanConroy** [comment]
I don't think VSCode really uses the .sln

**Comment by davidfowl** [comment]
It does, actutslly the more I think about it let’s move this to the samples repo.

**Comment by anurse** [comment]
It doesn't load in VS code. The rest of the projects work fine but you do get a warning. Closing this.

**Pull request closed without merging**